README.md

```diff
@@ -1,2 +1,52 @@
-# micro-rdk-module-template
-Template for micro-rdk modules
+# Viam Micro-RDK Module Template
+
+## (In)stability Notice
+
+> **Warning** The Viam Micro-RDK is currently in beta.
+
+## Overview
+
+This repository defines a template for use with
+[`cargo-generate`](https://cargo-generate.github.io/cargo-generate). Projects
+created from this template are a starting point for defining modular
+resources for the Micro-RDK.
+
+Modular resources for the Viam Micro-RDK function differently than
+modular resources for the standard Viam RDK, because they must be
+compiled into the image which will be flashed to the microcontroller.
+
+## Usage
+
+`cargo generate --git https://github.com/viamrobotics/micro-rdk-module-template`
+
+Answer the prompts, and then implement your modular resources within
+the newly formed project. The `register_models` function defined in
+`src/lib.rs` will be called for you when your robot project starts up,
+and you can register your newly defined models with the Micro-RDK by
+invoking the appropriate methods on the Micro-RDK `ComponentRegistry`
+argument.
+
+Once you have implemented your module, you can use it in your
+Micro-RDK robot project simply by adding it as a standard dependencies
+in the `[dependencies]` section of your robot project's `Cargo.toml`
+file. The `register_models` entry point of all dependencies produced
+by this template will be automatically invoked at startup.
+
+## Tutorial
+
+Please see the [ESP32 Sensors
```

Comment:
ah, the instructions are here - maybe a "read this doc in its entirety" right under the beta notice.